src/Network/Socket.php

```diff
@@ -161,8 +161,13 @@ public function connect()
         }
 
         set_error_handler([$this, '_connectionErrorHandler']);
-        $this->connection = stream_socket_client(
-            $scheme . $this->_config['host'] . ':' . $this->_config['port'],
+        $remoteSocketTarget = $scheme . $this->_config['host'];
+        $port = $this->_config['port'];
+        if (is_int($port)) {
+            $remoteSocketTarget .= ':' . (int)$port;
```

Comment:
Why the cast to int when already `is_int()` check earlier?  I still preferred the `!== null` check just incase someone sets the port as string.

Reply:
ok, I'll allow string but cast it to int to ensure we're dealing with an int port